core-bundle/src/ServiceAnnotation/AbstractFragmentAnnotation.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\ServiceAnnotation;
+
+use Terminal42\ServiceAnnotationBundle\Annotation\ServiceTag;
+
+abstract class AbstractFragmentAnnotation extends ServiceTag
```

Comment:
Is `AbstractFragmentAnnotation` the correct name here? Hooks and callbacks are not related to fragments I think. How about `AbstractServiceTagAnnotation`?

Reply:
Or do we need another parent class that doesn’t include `$category`, `$renderer` and `$template` and `AbstractFragmentAnnotation` can then inherit from that class?

Reply:
All fragment controllers have the same abstract. Hooks and callbacks are not fragment, they don‘t implement that class.

Reply:
But the code says  
`final class Callback extends AbstractFragmentAnnotation`  
and  
`final class Hook extends AbstractFragmentAnnotation`  
?

Reply:
You're absolutely right, fixed in 0613a4d515622a970dd3956339d284d40dd3ce19